File: C++/74. Search a 2D Matrix.cpp
class Solution {
public:
    bool searchMatrix(vector<vector<int>>& m, int t) {
        int i,j;
        for(i=0;i<m.size();i++)
        {
            if(m[i][0]<=t && m[i][m[0].size()-1]>=t)
            {
                for(j=0;j<m[0].size();j++)
                {
                    if(m[i][j]==t)
                    return true;
                }
            }
        }
        return false;
    }
};

File: JAVA/74. Search a 2D Matrix.java
class Solution {
    public boolean searchMatrix(int[][] m, int t) {
        int i,j,l=0,u=m[0].length-1,c;
        for(i=0;i<m.length;i++){
            if(m[i][m[0].length-1]>=t){
                while(l<=u){
                   c=(l+u)/2;
                   if(m[i][c]==t)
                   return true;
                   else if(m[i][c]>t)
                   u=--c;
                   else
                   l=++c;
                }
                return false;
            }
        }
        return false;
    }
}